src/frequently-asked-questions.md

```diff
@@ -1 +1,238 @@
+
+- Cargo provices some default profiles, we reccomend using the [`release` profile][release-profile] as it optimizes and removes debug symbols.
+- Cargo allows different [profile settings][profile-settings-cargo], which can make a difference in the resulting size of the artifact.
+  - See [Optimizations: the speed size tradeoff][embedded-book-tradeoffs] of The Embedded Rust Book.
+- Careful when using external dependencies, as they can increase the size of your resulting artifact
+- Filter log messages if they are not going to be useful or read.
+
+[embedded-book-tradeoffs]: https://docs.rust-embedded.org/book/unsorted/speed-vs-size.html
+[release-profile]: https://doc.rust-lang.org/cargo/reference/profiles.html#release
+[profile-settings-cargo]: https://doc.rust-lang.org/cargo/reference/profiles.html#profile-settings
+
+### Reducing Memory Usage
+
+- Minimize heap allocations
+- Optimize Struct Layouts to improve the padding
+
+## Using Crates from Git
+
+### Using a Crate Version from a Repository
+
+If for some reason you dont want to use a published version, you can use a Git repository directly. To specify a crate from a Git repository and a specific branch:
+```toml
+[dependencies]
+esp-hal = { git = "https://github.com/esp-rs/esp-hal.git", package="esp-hal", branch = "main" }
+```
+
+You can also specify a specific commit or tag:
+```toml
+[dependencies]
+esp-hal = { git = "https://github.com/esp-rs/esp-hal.git", package="esp-hal", rev = "abc123" }
+```
+ See [Specifying dependencies from git repositories][cargo-dependencies-git] section of The Cargo Book for more information.
+
+[cargo-dependencies-git]: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories
+
+### Appliying Patches to Dependencies
+
+If you need to override a dependency with a local or different version, use the `[patch]` section in `Cargo.toml`:
+```toml
+[patch.crates-io]
+esp-hal = { git = "https://github.com/esp-rs/esp-hal.git", branch = "main" }
+```
+
+For more information, see [`[patch]` section][cargo-patch] of The Cargo Book.
+
+[cargo-patch]: https://doc.rust-lang.org/cargo/reference/config.html?highlight=patches#patch
+
+## Simulating Projects
+
+Simulating projects can be handy. It allows users to test projects using CI, try projects without having hardware available, and many other scenarios.
+
+At the moment, there are a few ways of simulating Rust projects on Espressif chips. Every way has some limitations, but it's quickly evolving and getting better every day.
+
+In this chapter, we will discuss currently available simulation tools.
+
+Refer to the table below to see which chip is supported in every simulating method:
+
+|              | **[Wokwi][wokwi]** | **QEMU** |
+| :----------: | :----------------: | :------: |
+|  **ESP32**   |         ✅          |    ✅     |
+| **ESP32-C2** |         ❌          |    ❌     |
+| **ESP32-C3** |         ✅          |    ❌     |
```

Comment:
IIRC C3 is supported on QEMU?

Reply:
ESP32, C3, and S3 should be supported in qemu: https://github.com/espressif/esp-toolchain-docs/blob/main/qemu/README.md